### PR TITLE
Linear Elasticity Lattice Example

### PR DESCRIPTION
**Pull request by larsson4**
This PR introduces a lattice example for linear elasticity.

Tasks:
- [x] Create simple L shaped structure (meshes, config file and YAML)
- [x] L-Cantilever CWROM experiment
- [x] Lattice config generation
- [x] Quadrilateral truss sheared beam experiment

### CODE REVIEW

**Comment by larsson4** [comment]
To test the L-cantilever example

```
# Sample generation
../../bin/main -i linelast.simpleL.yml -f main/mode=sample_generation

# Training ROM
../../bin/main -i linelast.simpleL.yml -f main/mode=train_rom

# Building ROM
../../bin/main -i linelast.simpleL.yml -f main/mode=build_rom:main/use_rom=true

# Single run with ROM
../../bin/main -i linelast.simpleL.yml -f main/use_rom=true:visualization/enabled=true:visualization/unified_paraview=false

# Save comparison
../../bin/main -i linelast.simpleL.yml -f main/use_rom=true -o comparison.h5
```

**Comment by axla-io** [comment]
Test the lattice program:
```
# Sample generation
../../bin/main -i linelast.lattice.yml -f main/mode=sample_generation

# Training ROM
../../bin/main -i linelast.lattice.yml -f main/mode=train_rom

# Building ROM
../../bin/main -i linelast.lattice.yml -f main/mode=build_rom:main/use_rom=true

# Single run with ROM
../../bin/main -i linelast.lattice.yml -f main/use_rom=true:visualization/enabled=true:visualization/unified_paraview=false

# Save comparison
../../bin/main -i linelast.lattice.yml -f main/use_rom=true -o comparison.h5
```

**Comment by axla-io** [comment]
@dreamer2368 With the latest commit the program runs and the lattice structure seems to be generated correctly, However, the relative error is high. Do you think that it's an implementation issue?

**Comment by dreamer2368** [comment]
> @dreamer2368 With the latest commit the program runs and the lattice structure seems to be generated correctly, However, the relative error is high. Do you think that it's an implementation issue?

The relative error can vary depending on number of samples and how you set up sample problems/parameters, and any miscellaneous setups we missed. I don't have any necessary relevant information here, which would be better discussed on a separate meeting.

**Comment by axla-io** [comment]
@dreamer2368 I fixed the problem and now the lattice example runs!

File: src/linelast_solver.cpp [hunk 1]
```diff
@@ -55,7 +55,11 @@ LinElastSolver::~LinElastSolver()
    delete globalMat;
    delete globalMat_hypre;
    delete mumps;
+   if (init_x != NULL)
```
Comment (dreamer2368) [inline src/linelast_solver.cpp]: `delete init_x;` shouldn't have a problem as long as `init_x` is initialized as `NULL` in `include/linelast_solver.hpp`. Initialize it with `NULL` and remove the if clause.

File: src/linelast_solver.cpp [hunk 2]
```diff
@@ -133,6 +137,7 @@ void LinElastSolver::InitVariables()
    }
    if (use_rom)
      MultiBlockSolver::InitROMHandler();
+   init_x = NULL;
```
Comment (dreamer2368) [inline src/linelast_solver.cpp]: This could be included in `include/linelast_solver.hpp`, as mentioned above

File: src/parameterized_problem.cpp [hunk 3]
```diff
@@ -179,6 +179,11 @@ void init_disp(const Vector &x, Vector &u)
    u(u.Size()-1) = -0.2*x(0)*rdisp_f;
 }
 
+void init_disp_lcantilever(const Vector &x, Vector &u)
+{
+   u = 0.0;
+   u(u.Size()-1) = -0.2*(x(u.Size()-1) - 5.0)*rdisp_f;
```
Comment (dreamer2368) [inline src/parameterized_problem.cpp:185]: This comment doesn't have to be in this PR, just for a suggestion: I guess in the future we could add more parameters for `LineElastDispCantilever` that control `-0.2` and `-5.0` values here?

File: utils/python/linelast_comp.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        # 3-dimension vector for translation / rotation
+        grp.create_dataset("configuration", mesh_configs.shape, data=mesh_configs)
+
+        grp = f.create_group("ports")
+        grp.attrs["number_of_references"] = 2
+        grp.attrs["0"] = "port1"
+        grp.attrs["1"] = "port2"
+        grp.create_dataset("interface", if_data.shape, data=if_data)
+
+        port = grp.create_group("port1")
```
Comment (dreamer2368) [inline utils/python/linelast_comp.py:60]: Thanks for adding this example.

**Review by dreamer2368 — APPROVED** [review]
Added a minor comment about `init_x`. It mostly looks good.

**Comment by dreamer2368** [comment]
One request of change: update `README.md` with linear elasticity as supported physics, if possible.

File: src/parameterized_problem.cpp (hunk 3 above)
Comment (axla-io) [reply src/parameterized_problem.cpp:185]: Yes! The -5 value should not be hardcoded but correspond to w + h, where w and h are the width and length of the rod component respectively

**Pull request merged**